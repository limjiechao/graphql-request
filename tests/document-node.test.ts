import { request } from '../src/index.js'
import { setupMockServer } from './__helpers.js'
import { gql } from 'graphql-tag'
import { expect, it } from 'vitest'

const ctx = setupMockServer()

it(`accepts graphql DocumentNode as alternative to raw string`, async () => {
  const mock = ctx.res({ body: { data: { foo: 1 } } })
  await request(
    ctx.url,
    gql`
      {
        query {
          users
        }
      }
    `,
  )
  expect(mock).toMatchSnapshot()
})
