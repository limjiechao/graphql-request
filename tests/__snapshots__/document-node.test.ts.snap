// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`accepts graphql DocumentNode as alternative to raw string 1`] = `
{
  "requests": [
    {
      "body": {
        "query": "{
  query {
    users
  }
}",
      },
      "headers": {
        "accept": "application/graphql-response+json, application/json",
        "accept-encoding": "gzip, deflate",
        "accept-language": "*",
        "connection": "keep-alive",
        "content-length": "43",
        "content-type": "application/json",
        "host": "DYNAMIC",
        "sec-fetch-mode": "cors",
        "user-agent": "node",
      },
      "method": "POST",
    },
  ],
  "spec": {
    "body": {
      "data": {
        "foo": 1,
      },
    },
  },
}
`;
